File: scripts/generate-CITATION.cff.py
#!/usr/bin/env python
"""Generate ``CITATION.cff``."""
import sys

try:
    import tomllib  # type: ignore
except ImportError:
    import tomli as tomllib

with open(sys.argv[1], "r") as f:
    citation = f.read()

with open(sys.argv[2], "rb") as f:
    project = tomllib.load(f)["project"]

print(
    citation.format(
        url=project["urls"]["Source"],
        name=project["name"],
        description=project["description"],
    ),
    end="",
)

File: scripts/generate-CITATION.cff.pl
#!/usr/bin/env -S perl -pi
unless ($#ARGV < 0) {
  $name = $1 if /^name = "([^"]*+)"/;
  $description = $1 if /^description = "([^"]*+)"/;
  $url = $1 if /^Source = "([^"]*+)"/;
  next;
};
s/(?<=^title: ")([^"]*+)(?="$)/$name: $description/;
s/(?<=^url: ")([^"]*+)(?="$)/$url/;
